src/bin/render-readmes.rs

```
@@ -0,0 +1,330 @@
+        "[{}-{}] Invalid tar archive entries",
+        version.krate,
+        version.num
+    ));
+    let manifest: Manifest = {
+        let path = format!("{}-{}/Cargo.toml", version.krate, version.num);
+        let contents = find_file_by_path(&mut entries, Path::new(&path), &version).unwrap();
+        toml::from_str(&contents).expect(&format!(
+            "[{}-{}] Syntax error in manifest file",
+            version.krate,
+            version.num
+        ))
+    };
+    if manifest.package.readme.is_none() {
+        return None;
+    }
+    let rendered = {
+        let path = format!(
+            "{}-{}/{}",
+            version.krate,
+            version.num,
+            manifest.package.readme.unwrap()
+        );
+        let contents = find_file_by_path(&mut entries, Path::new(&path), &version).unwrap();
+        markdown_to_html(&contents).expect(&format!(
+            "[{}-{}] Couldn't render README",
+            version.krate,
+            version.num
+        ))
+    };
+    return Some(rendered);
+    #[derive(Deserialize)]
+    struct Package {
+        readme: Option<String>,
+    }
+    #[derive(Deserialize)]
+    struct Manifest {
+        package: Package,
+    }
+}
+
+/// Search an entry by its path in a Tar archive.
+fn find_file_by_path<R: Read>(
+    mut entries: &mut tar::Entries<R>,
+    path: &Path,
+    version: &EncodableVersion,
+) -> Option<String> {
+    let mut file = entries
+        .find(|entry| match *entry {
+            Err(_) => false,
+            Ok(ref file) => {
+                let filepath = match file.path() {
+                    Ok(p) => p,
+                    Err(_) => return false,
+                };
+                return filepath == path;
+            }
+        })
+        .expect(&format!(
+            "[{}-{}] file is not present: {}",
+            version.krate,
+            version.num,
+            path.display()
+        ));
+    match file {
```

LOL this code was really confusing to me until I looked at the `tar` docs and saw that `tar::Entries` iterates over `Result<Entry>` 😂😂😂

Since everywhere else in `find_file_by_path` is calling `expect` and crashing, wdyt about calling `expect` instead of this `match` and making this method return just `String` rather than `Option<String>`, and removing the `unwrap` on line 200? 

I'm not entirely sure what the error would be here, so I don't know what to *say* in the `expect`. I was going to suggest making the `unwrap` on line 200 be an `expect` that says "file not found" but I see that there's already an `expect` that says just that in this method!!